psydac/fem/projectors.py

```diff
@@ -158,3 +160,335 @@ def get_dual_dofs(Vh, f, domain_h, backend_language="python", return_format='ste
         return tilde_f.toarray()
     else:
         return tilde_f
+
+
+#===============================================================================
+class BoundaryProjector(LinearOperator):
+    """
+    A LinearOperator that applies homogeneous (unless manually given different bcs) Dirichlet boundary conditions.
+
+    Parameters
+    ----------
+    fem_space : psydac.fem.basic.FemSpace
+        fem_space.coeff_space is domain and codomain of this LO. fem_space.kind.name determines the BCs that get applied.
+
+    bcs : Iterable | None
+        Iterable of sympde.topology.Boundary objects. 
+        Allows the user to apply different kinds of BCs.
+        Must not be passed if a space_kind argument is passed.
+
+    space_kind : str | SpaceType | None
+        Necessary only if fem_space.kind.name is undefined and no bcs are passed.
+        Must not be passed if a space_kind argument is passed.
+
+    Notes
+    -----
+    See examples/vector_potential_3d.py for a use case of such an operator.
+    
+    """
+    def __init__(self, fem_space, *, bcs=None, space_kind=None):
+
+        assert isinstance(fem_space, FemSpace)
+        assert bcs is None or isinstance(bcs, Iterable)
+        assert space_kind is None or isinstance(space_kind, (str, SpaceType))
+        if bcs is not None:
+            assert space_kind is None
+        if space_kind is not None:
+            assert bcs is None
+
+        coeff_space    = fem_space.coeff_space
+        self._domain   = coeff_space
+        self._codomain = coeff_space
+
+        if bcs is not None:
+            assert all(isinstance(bc, Boundary) for bc in bcs)
+            self._bcs = bcs
```

Comment:
Why have you changed this line back to the original version? The user may pass any iterable object, but we want to always store the attribute `_bcs` as a tuple:
```suggestion
            self._bcs = tuple(bcs)
```